src/SFA.DAS.EAS.Api/Orchestrators/AccountsOrchestrator.cs

```diff
@@ -44,11 +43,88 @@ public AccountsOrchestrator(IMediator mediator, ILogger logger )
+                data.Add(new AccountWithBalanceViewModel { AccountId = account.Id, AccountName = account.Name, AccountHashId = account.HashedId, Balance = accountBalance?.Balance ?? 0 });
             });
 
             return new OrchestratorResponse<PagedApiResponseViewModel<AccountWithBalanceViewModel>>() {Data = new PagedApiResponseViewModel<AccountWithBalanceViewModel>() {Data = data, Page = pageNumber, TotalPages = (accountsResult.AccountsCount / pageSize) + 1} };
         }
 
+        public async Task<OrchestratorResponse<AccountDetailViewModel>> GetAccount(string hashedAccountId)
+        {
+            var accountResult = await _mediator.SendAsync(new GetEmployerAccountByHashedIdQuery { HashedAccountId = hashedAccountId });
+            if (accountResult.Account == null)
+            {
+                return new OrchestratorResponse<AccountDetailViewModel> { Data = null };
+            }
+
+            var viewModel = ConvertAccountDetailToViewModel(accountResult);
+            return new OrchestratorResponse<AccountDetailViewModel> { Data = viewModel };
+        }
+
+        public async Task<OrchestratorResponse<LegalEntityViewModel>> GetLegalEntity(long legalEntityId)
+        {
+            var legalEntityResult = await _mediator.SendAsync(new GetLegalEntityByIdQuery { Id = legalEntityId });
+            if (legalEntityResult.LegalEntity == null)
+            {
+                return new OrchestratorResponse<LegalEntityViewModel> { Data = null };
+            }
+
+            var viewModel = ConvertLegalEntityToViewModel(legalEntityResult);
+            return new OrchestratorResponse<LegalEntityViewModel> { Data = viewModel };
+        }
+
+        public async Task<OrchestratorResponse<PayeSchemeViewModel>> GetPayeScheme(string payeSchemeRef)
+        {
+            var payeSchemeResult = await _mediator.SendAsync(new GetPayeSchemeByRefQuery { Ref = payeSchemeRef });
+            if (payeSchemeResult.PayeScheme == null)
+            {
+                return new OrchestratorResponse<PayeSchemeViewModel> { Data = null };
+            }
+
+            var viewModel = ConvertPayeSchemeToViewModel(payeSchemeResult);
+            return new OrchestratorResponse<PayeSchemeViewModel> { Data = viewModel };
+        }
+
+        private PayeSchemeViewModel ConvertPayeSchemeToViewModel(GetPayeSchemeByRefResponse payeSchemeResult)
+        {
+            var payeSchemeViewModel = new PayeSchemeViewModel
+            {
+                Name = payeSchemeResult.PayeScheme.Name,
+                Ref = payeSchemeResult.PayeScheme.Ref
+            };
+
+            return payeSchemeViewModel;
+        }
+
+        private LegalEntityViewModel ConvertLegalEntityToViewModel(GetLegalEntityByIdResponse legalEntityResult)
+        {
+            var legalEntityViewModel = new LegalEntityViewModel
+            {
+                DateOfIncorporation = legalEntityResult.LegalEntity.DateOfIncorporation,
```

Comment:
Just a quick note, we use automapper in the projects so you can use this for your mapping to save on extra code. We are aware some people don't like to use it which is fine however its there and being used if you wish to use it. It will save you quite a bit of time for simple mapping